src/lib/blockly/blocks/exact.ts

```diff
@@ -5,13 +5,13 @@ export default {
     "message0": "exactly %1",
     "args0": [
       {
-        "type": "input_value",
-        "name": "PROPOSITION",
-        "check": Proposition,
+        "type": "field_input",
```

Comment:
when copy-pasting templates, be careful to separate the stuff we're using for statements (variables, combinations...) and for proofs (intro, destruct...)